### DISCUSSION TITLE
How to Obfuscate Codes in Nuxt 3?

### DISCUSSION THREAD

**Discussion by Fiyuva** [nuxt/nuxt#25948] [Questions]
I have a project that I coded with nuxt 3. I want to do Code Obfuscation, but I couldn't find a proper resource. There are .ts and .vue files in my project. Can you help me?

I couldn't try anything because I couldn't find a proper source.

**Comment by Fiyuva** [comment]
any update?

**Comment by huang-julien** [comment]
You probably should take a look a Vite plugins. https://github.com/elmeet/vite-plugin-javascript-obfuscator

But build code should be already minified.

**Reply by Fiyuva** [reply]
Why did you choose vite instead of webpack to obfuscate?

**Reply by AloisSeckar** [reply]
Vite is now prefered build tool in Vue 3 - and thus Nuxt 3. It is still possible to use webpack, but it tend to be a legacy technology now. So most people working with Nuxt 3 would naturally seek Vite-based solutions.

**Reply by Fiyuva** [reply]
`const opoptions = {
  compact: true,
  controlFlowFlattening: true,
  controlFlowFlatteningThreshold: 0.75,
  deadCodeInjection: true,
  deadCodeInjectionThreshold: 0.4,
  debugProtection: true,
  debugProtectionInterval: 3000,
  disableConsoleOutput: true,
  domainLock: [],
  domainLockRedirectUrl: "about:blank",
  forceTransformStrings: [],
  identifierNamesCache: null,
  identifierNamesGenerator: "hexadecimal",
  identifiersDictionary: [],
  identifiersPrefix: "",
  ignoreImports: false,
  inputFileName: "Fiyuva",
  log: false,
  numbersToExpressions: true,
  optionsPreset: "high-obfuscation",
  renameGlobals: true,
  renameProperties: true,
  renamePropertiesMode: "safe",
  reservedNames: [],
  reservedStrings: [],
  seed: 0,
  selfDefending: false,
  simplify: true,
  sourceMap: true,
  sourceMapBaseUrl: "",
  sourceMapFileName: "",
  sourceMapMode: "separate",
  sourceMapSourcesMode: "sources-content",
  splitStrings: true,
  splitStringsChunkLength: 10,
  stringArray: true,
  stringArrayCallsTransform: true,
  stringArrayCallsTransformThreshold: 0.75,
  stringArrayEncoding: ["rc4"],
  stringArrayIndexesType: ["hexadecimal-number"],
  stringArrayIndexShift: true,
  stringArrayRotate: true,
  stringArrayShuffle: true,
  stringArrayWrappersCount: 2,
  stringArrayWrappersChainedCalls: true,
  stringArrayWrappersParametersMaxCount: 4,
  stringArrayWrappersType: "function",
  stringArrayThreshold: 0.75,
  target: "browser",
  transformObjectKeys: true,
  unicodeEscapeSequence: false
};`

`obfuscatorPlugin({
        include: [/\.(js|ts|cjs|mjs)$/],
        exclude: [/node_modules/, /\.nuxt/],
        debugger: true,
        options: { opoptions }
      })`


First of all, thank you for your answers. I made a configuration like this. Can you see an error? If there is no error, what is the reason why I could not obfuscate the html and css codes?

**Reply by wys010203** [reply]
"Hydration completed but contains mismatches" after obfuscate

**Comment by Fiyuva** [comment]
??
